gstreamer/src/event.rs

```diff
@@ -1667,3 +1667,50 @@ impl<'a> CustomBothOobBuilder<'a> {
         ev
     });
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_simple() {
+        ::init().unwrap();
+
+        // Event without arguments
+        let flush_start_evt = Event::new_flush_start().build();
+        match flush_start_evt.view() {
+            EventView::FlushStart(flush_start_evt) => {
+                assert!(flush_start_evt.0.get_structure().is_none());
+            },
+            _ => panic!("flush_start_evt.view() is not an EventView::FlushStart(_)"),
+        }
+
+        let flush_start_evt = Event::new_flush_start()
+            .other_fields(&[("extra-field", &true)])
+            .build();
+        match flush_start_evt.view() {
+            EventView::FlushStart(flush_start_evt) => {
+                assert!(flush_start_evt.0.get_structure().is_some());
+                if let Some(other_fields) = flush_start_evt.0.get_structure() {
+                    assert!(other_fields.has_field("extra-field"));
+                }
+            },
+            _ => panic!("flush_start_evt.view() is not an EventView::FlushStart(_)"),
+        }
+
+        // Event with arguments
+        let flush_stop_evt = Event::new_flush_stop(true)
+            .other_fields(&[("extra-field", &true)])
+            .build();
+        match flush_stop_evt.view() {
+            EventView::FlushStop(flush_stop_evt) => {
+                assert_eq!(flush_stop_evt.get_reset_time(), true);
+                assert!(flush_stop_evt.0.get_structure().is_some());
```

Comment:
What would you think of implementing a `deref` on the concret events & messages in order to get direct access to the `EventRef` & `MessageRef` methods without using `.0`?

Reply:
Yes, I thought I did that already. `.0` should also be private.

Please add that if you want, same for event and query if needed :)

Reply:
Also sorry for not replying anything on the TOC bug so far, it needs some harder thinking by Tim and myself and we didn't get to it yet. If you're coming to FOSDEM, maybe we can discuss there?

It's on my list in any case

Reply:
OK! Thank you for the feedback. I'm not coming unfortunately. I can open a thread on the mailing list if that helps.

Reply:
Not really, it's mostly a matter of finding time to discuss and think it through :)